msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerAuthServiceStrategy.java

```diff
@@ -309,4 +272,79 @@ static Bundle getRequestBundleForRemoveAccount(@NonNull final OperationParameter
 
         return requestBundle;
     }
+
+    @WorkerThread
+    protected boolean getDeviceMode(@NonNull OperationParameters parameters) throws BaseException, InterruptedException, ExecutionException, RemoteException {
+        return performAuthServiceOperation(parameters.getAppContext(),
+                new AuthServiceOperation<Boolean>() {
+                    @Override
+                    public Boolean perform(IMicrosoftAuthService service) throws BaseException, RemoteException {
+                        return MsalBrokerResultAdapter
+                                .deviceModeFromBundle(
+                                        service.getDeviceMode()
+                                );
+                    }
+
+                    @Override
+                    public String getOperationName() {
+                        return ":getDeviceModeWithAuthService";
+                    }
+                });
+    }
+
+    @WorkerThread
+    protected List<ICacheRecord> getCurrentAccountInSharedDevice(@NonNull final OperationParameters parameters) throws InterruptedException, ExecutionException, RemoteException, BaseException {
+        return performAuthServiceOperation(parameters.getAppContext(),
+                new AuthServiceOperation<List<ICacheRecord>>() {
+                    @Override
+                    public List<ICacheRecord> perform(IMicrosoftAuthService service) throws RemoteException, BaseException {
+                        return MsalBrokerResultAdapter
+                                .accountsFromBundle(
+                                        service.getCurrentAccount(
+                                                BrokerAuthServiceStrategy.getRequestBundleForGetAccounts(parameters)
+                                        ));
+                    }
+
+                    @Override
+                    public String getOperationName() {
+                        return ":getCurrentAccountInSharedDeviceWithAuthService";
+                    }
+                });
+    }
+
+    @WorkerThread
+    protected void signOutFromSharedDevice(@NonNull final OperationParameters parameters) throws BaseException, InterruptedException, ExecutionException, RemoteException {
+        final String methodName = ":signOutFromSharedDeviceWithAuthService";
```

Comment:
`methodName` does not match the actual method name as it has `WithAuthService` appended.

Reply:
This is intentional, as every other function has "WithAuthService" appended. (I guess it's for debugging purpose - so that we know what subclasses to look for).